### PR TITLE
dev-db/cdb: fix some K&R decls and strict-prototypes

### PR DESCRIPTION
**Pull request by sevz17**

### CODE REVIEW

**Comment by sevz17** [comment]
It compiles, but it's not complete

File: dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+ {
+   unsigned long u;
+   u = 1;
+--- a/x86cpuid.c
++++ b/x86cpuid.c
+@@ -1,11 +1,11 @@
+-#include <signal.h>
+-
+-void nope()
++__END_DECLS void
```
Comment (thesamesam) [inline dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch:440]: this _END_DECLS thing looks suspicious, what's that about?

File: dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
++main (void)
+ {
+ #ifdef NeXT
+   printf("nextstep\n"); exit(0);
+--- a/tryulong32.c
++++ b/tryulong32.c
+@@ -1,4 +1,5 @@
+-void main()
++void 
++main (void)
```
Comment (thesamesam) [inline dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch:430]: main with non-int return type?

Comment (sevz17) [reply dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch:430]: Yeah, the whole file:
```c
void 
main (void)
{
  unsigned long u;
  u = 1;
  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
  if (!u) _exit(0);
  _exit(1);
}
```

Comment (sevz17) [reply dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch:430]: The _exit calls can be replaced by proper `return`s, right?

File: dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch (hunk 1 above)
Comment (sevz17) [reply dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch:440]: cproto kindly added it and I didn't realize, dropping

File: dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch (hunk 2 above)
Comment (thesamesam) [reply dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch]: main actually has special semantics: https://stackoverflow.com/questions/204476/what-should-main-return-in-c-and-c (no need for explicit return)

... but yes, we can do that change you suggest, and it's simpler if we do

**Pull request closed without merging**